### PR TITLE
Enforce FusedLayerNorm is ordered last

### PR DESCRIPTION
**Pull request by alextrott16**
Adds a single line of logic into `composer.core.engine` so that `FusedLayerNorm` is last in the algorithm execution order.

This will ensure that any `torch.nn.LayerNorm` modules which are added by other model surgery algorithms are caught by `FusedLayerNorm` and converted into the fused versions.

As a motivating use case, when `GatedLinearUnits` and `FusedLayerNorm` are used together on a BERT model, having `GatedLinearUnits` perform surgery last will result in a model that is roughly 50/50 fused and unfused layernorm instances.

### CODE REVIEW

File: composer/core/engine.py [hunk 1]
```diff
@@ -321,12 +321,15 @@ def _compile(
...
         """
-        from composer.algorithms import CutMix, MixUp, SelectiveBackprop, StochasticDepth
+        from composer.algorithms import CutMix, MixUp, SelectiveBackprop, StochasticDepth, FusedLayerNorm
 
         # Move selective backprop to the beginning while maintaining order of other algorithms
         algorithms = sorted(algorithms_to_run,
                             key=lambda x: not isinstance(x, SelectiveBackprop) and not isinstance(x, StochasticDepth))
 
+        # Move fused layernorm to the end while maintaining order of other algorithms (FLN only does surgery on leaf modules)
+        algorithms = sorted(algorithms_to_run, key=lambda x: isinstance(x, FusedLayerNorm))
```
Comment (ravi-mosaicml) [inline composer/core/engine.py:331]: I presume we want to keep the previous sort from line 327?

```suggestion
        algorithms = sorted(algorithms, key=lambda x: isinstance(x, FusedLayerNorm))
```

**Review by ravi-mosaicml — APPROVED** [review]
LGTM; see comment.

**Comment by ravi-mosaicml** [comment]
At some point I think we'll need to write a DAG scheduler for the algorithms....but that's a future problem :)

File: composer/core/engine.py (hunk 1 above)
Comment (alextrott16) [reply composer/core/engine.py:331]: Yes, thank you!

**Review by hanlint — APPROVED** [review]
LGTM
